code/modules/mob/living/living.dm

```diff
@@ -913,6 +913,9 @@
 		to_chat(G, "<span class='holoparasite'>Your summoner has changed form!</span>")
 
 /mob/living/rad_act(amount)
+	if(has_trait(TRAIT_RADIMMUNE)) //don't do anything if the mob is immune to radiation.
+		return
```

Comment:
Should this return after `. = ..()` instead? It's calling back to `SendSignal(COMSIG_ATOM_RAD_ACT)`. Don't think it would matter to send the signal if they're immune.